gitpod-io/gitpod
# Fix handling multiple repos with same context URL

## Description

When you have a Repository setup like so:
- Repo A (github.com/filiptronicek/samerepo)
- Repo B (github.com/filiptronicek/samerepo)

The new workspace modal does not like that and when you pick Repo B, it picks Repo A instead. It was just a types mismatch.

Additionally, there were issues with workspaces contributing deleted projects and contributing projects without names. This also fixes a bug in the "Add Repository" modal, where importing an already configured repository resulted in showing the name of the existing configuration instead of the source git repo.

## Related Issue(s)

Fixes EXP-1549 EXP-1550 EXP-1521

## How to test

1. Import 1 repository twice and distinguish the copies by name
2. Open the new workspace page and search for the repo
3. Select the first one, make sure the thing you clicked on stayed as the active item in the selection dropdown. Then, do the same for the second Repo.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-fix-mul695419239a</li>
	<li><b>🔗 URL</b> - <a href="https://ft-fix-mul695419239a.preview.gitpod-dev.com/workspaces" target="_blank">ft-fix-mul695419239a.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-fix-multiple-repos-per-git-repo-gha.23541</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-fix-mul695419239a%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
